v2/yarpcthrift/thriftrw-plugin-yarpc2/fx_test.go

```diff
@@ -144,9 +145,12 @@ func TestFxServer(t *testing.T) {
 		_, err := client.Integer(ctx, ptr.String("baz")) // baz does not exist
 		assert.Error(t, err, "request failed")
 
-		exc, ok := err.(*atomic.KeyDoesNotExist)
-		require.True(t, ok, "error '%+v' must be a *KeyDoesNotExist, not %T", err, err)
-		assert.Equal(t, "baz", *exc.Key, "exception key did not match")
+		// TODO(mhp): After we have a way to map errors between YARPC errors
```

Comment:
In the future, what do we expect the test to look like, assuming that the returned result is a well-defined thrift exception?

Reply:
The test should go back to what it looked like before my change. Currently, the error returning to the client is being propagated down to the encoding layer as a `yarpcerror.Status`; instead the transport layer should handle some sort of error decoding before it sends it to the encoding layer. Hopefully, the error decoding part successfully decodes it to a specific thrift exception, but not too sure yet if this will be possible.